adoc/chapters/device_compiler.adoc

```diff
@@ -219,24 +219,6 @@ The standard {cpp} fixed width types, e.g. [code]#int8_t#,
 should have the same size as defined by the {cpp} standard for host and
 device.
 
-ifdef::showtodos[]
-[NOTE]
-.Note
-====
-To ensure that pointer types and [code]#size_t# use the same amount of
-storage on host and device when inside structures, SYCL also requires that
-device compilers use the host sizes for pointers or [code]#size_t#. Devices
-may be using a smaller size internally for pointers and [code]#size_t#,
-but this should not impact the programming model for users. In the case where
-devices use larger pointer size internally than the host, then
-[code]#size_t# will match the <<host-pointer>> size. For structures shared
-between host and device it is recommended to use fixed width types for pointers.
-Scalar data types for a SYCL device compiler are described in
-<<table.types.fundamental>>.
-====
-endif::showtodos[]
```

Comment:
I forgot that we had this description.
It looks like more as an implementation note.
Do you think this has to be removed?

Reply:
I suggested removing this note in #312.  The first part of this note reads like a suggestion for implementers, but the second to last sentence sounds like a requirement for applications:

> For structures shared between host and device it is recommended to use fixed width types for pointers.

That is definitely not true.  We do not want to force applications to use the fixed-width types for all structures shared between host and device.

Personally, I don't think this note is helpful to implementers.  We already say that fundamental types have the same size on host and device, so this statement about `size_t` is redundant.  The statement about devices using a larger pointer size also seems confusing to me.  I was just worried that this note added more confusion than clarity.